### PR TITLE
S06 us25 02 admin crud company

### PR DESCRIPTION
**Pull request by RedPandore**

### CODE REVIEW

**Review by bouboumael — DISMISSED** [review]

**Review by bouboumael — DISMISSED** [review]

File: src/Controller/CompanyController.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            $entityManager->flush();
+
+            return $this->redirectToRoute('company_index');
+        }
+
+        return $this->render('company/new.html.twig', [
+            'company' => $company,
+            'form' => $form->createView(),
+        ]);
+    }
```
Comment (LucyMerlier) [inline src/Controller/CompanyController.php:51]: so an admin can create a new company?

File: src/Controller/CompanyController.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            $this->getDoctrine()->getManager()->flush();
+
+            return $this->redirectToRoute('company_index');
+        }
+
+        return $this->render('company/edit.html.twig', [
+            'company' => $company,
+            'form' => $form->createView(),
+        ]);
+    }
```
Comment (LucyMerlier) [inline src/Controller/CompanyController.php]: so an admin can edit a company's informations?

File: src/Form/CompanyType.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+class CompanyType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder
+            ->add('name')
+            ->add('address')
+            ->add('code')
+        ;
```
Comment (LucyMerlier) [inline src/Form/CompanyType.php]: no customized labels? Types?

File: templates/company/_delete_form.html.twig [hunk 4]
```diff
@@ -0,0 +1,4 @@
+<form method="post" action="{{ path('company_delete', {'id': company.id}) }}" onsubmit="return confirm('Are you sure you want to delete this item?');">
```
Comment (LucyMerlier) [inline templates/company/_delete_form.html.twig]: confirm message should be in french

File: templates/company/_form.html.twig [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{{ form_start(form) }}
+    {{ form_widget(form) }}
+    <button class="btn btn-primary">{{ button_label|default('Save') }}</button>
```
Comment (LucyMerlier) [inline templates/company/_form.html.twig]: button should be in french

File: templates/company/edit.html.twig [hunk 6]
```diff
@@ -0,0 +1,23 @@
+{% extends 'base.html.twig' %}
```
Comment (LucyMerlier) [inline templates/company/edit.html.twig]: this extends base directly? not a layout?

File: templates/company/index.html.twig [hunk 7]
```diff
@@ -0,0 +1,61 @@
+{% extends 'base.html.twig' %}
```
Comment (LucyMerlier) [inline templates/company/index.html.twig]: this extends base directly? not a layout?

File: templates/company/index.html.twig [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+                            <td>{{ company.address }}</td>
+                            <td>{{ company.code }}</td>
+                            <td>
+                                <a href="{{ path('company_show', {'id': company.id}) }}" class="btn fas fa-eye text-primary"></a>
+                                <a href="{{ path('company_edit', {'id': company.id}) }}" class="fas fa-pencil-alt"></a>
+                            </td>
+                        </tr>
+                    {% else %}
+                        <tr>
+                            <td colspan="5">Il n'y a aucune entreprises</td>
```
Comment (LucyMerlier) [inline templates/company/index.html.twig]: entreprise => singular, no "s"

File: templates/company/new.html.twig [hunk 9]
```diff
@@ -0,0 +1,11 @@
+{% extends 'base.html.twig' %}
```
Comment (LucyMerlier) [inline templates/company/new.html.twig]: this extends base directly? not a layout?

File: templates/company/show.html.twig [hunk 10]
```diff
@@ -0,0 +1,37 @@
+{% extends 'base.html.twig' %}
```
Comment (LucyMerlier) [inline templates/company/show.html.twig]: this extends base directly? not a layout?

File: templates/company/new.html.twig [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+
+{% block title %}New Company{% endblock %}
+
+{% block body %}
+    <h1>Create new Company</h1>
+
+    {{ include('company/_form.html.twig') }}
+
+    <a href="{{ path('company_index') }}">back to list</a>
+{% endblock %}
```
Comment (LucyMerlier) [inline templates/company/new.html.twig]: whole document should be in french

File: templates/company/show.html.twig [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+                            {{ company.id }}</p>
+                    </div>
+                </div>
+                <a href="{{ path('company_index') }}" class="btn btn-primary m-2">Retour à la liste</a>
+                <a href="{{ path('company_edit', {'id': company.id}) }}" class="btn btn-primary m-2">Éditer</a>
+                {{ include('company/_delete_form.html.twig') }}
+            </div>
+        </div>
+    </div>
+{% endblock %}
```
Comment (LucyMerlier) [inline templates/company/show.html.twig]: whole document should be in french

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
see comments

File: src/Controller/CompanyController.php (hunk 1 above)
Comment (RedPandore) [reply src/Controller/CompanyController.php:51]: Yes, the company ask JobPermut to get an access

**Review by bouboumael — DISMISSED** [review]

File: src/Form/CompanyType.php [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+        $builder
+            ->add('name', TextType::class, [
+                'attr' => ['placeholder' => 'JobPermut SARL'],
+                'label' => 'Nom de votre entreprise'
+                ])
+            ->add('address', TextType::class, [
+                'attr' => ['placeholder' => 'Orléans'],
+                'label' => 'Ville de l\'entreprise'
+                ])
+            ->add('code', NumberType::class, [
```
Comment (LucyMerlier) [inline src/Form/CompanyType.php:25]: Type number? do you really have an interest putting plus and minus arrows on this field?

**Review by LucyMerlier — APPROVED** [review]

File: src/Form/CompanyType.php (hunk 13 above)
Comment (LucyMerlier) [inline src/Form/CompanyType.php:25]: Type number? do you really have an interest putting plus and minus arrows on this field?

**Review by LucyMerlier — CHANGES_REQUESTED** [review]

File: src/Form/CompanyType.php (hunk 13 above)
Comment (RedPandore) [reply src/Form/CompanyType.php:25]: I've test it and I didn't see any plus and minus arrow. Test on Mozila and Chromium

**Review by jaldabaoth-code — DISMISSED** [review]

File: templates/company/edit.html.twig [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+    </div>
+    <div class="container">
+        <div class="col-2">
+            {{ include('company/_form.html.twig', {'button_label': 'Mettre à jour'}) }}
+
+            {{ include('company/_delete_form.html.twig') }}
+        </div>
+        <a href="{{ path('company_index') }}" class="btn btn-primary">Retourner à la liste</a>
+    </div>
+{% endblock %}
```
Comment (LucyMerlier) [inline templates/company/edit.html.twig]: maybe should not be there then

File: templates/company/index.html.twig [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+                <tbody>
+                    {% for company in companies %}
+                        <tr>
+                            <td>{{ company.id }}</td>
+                            <td>{{ company.name }}</td>
+                            <td>{{ company.address }}</td>
+                            <td>{{ company.code }}</td>
+                            <td>
+                                <a href="{{ path('company_show', {'id': company.id}) }}" class="btn fas fa-eye text-primary"></a>
+                                <a href="{{ path('company_edit', {'id': company.id}) }}" class="fas fa-pencil-alt"></a>
```
Comment (LucyMerlier) [inline templates/company/index.html.twig]: maybe should not be there then

File: templates/company/show.html.twig [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+                            {{ company.address }}
+                        </h5>
+                        <p class="m-0">Code :
+                            {{ company.code }}</p>
+                        <p class="m-0">Id :
+                            {{ company.id }}</p>
+                    </div>
+                </div>
+                <a href="{{ path('company_index') }}" class="btn btn-primary m-2">Retour à la liste</a>
+                <a href="{{ path('company_edit', {'id': company.id}) }}" class="btn btn-primary m-2">Éditer</a>
```
Comment (LucyMerlier) [inline templates/company/show.html.twig]: maybe should not be here then

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
see comments

**Review by bouboumael — APPROVED** [review]

**Review by LucyMerlier — APPROVED** [review]

**Pull request merged**